chapter_crashcourse/ndarray.md

```diff
@@ -1,22 +1,22 @@
+It is impossible to get anything done if we cannot manipulate data. Generally, there are two important things we need to do with data: (i) acquire it and (ii) process it once it is inside the computer. There is no point in acquiring data if we do not even know how to store it, so let's get our hands dirty first by playing with synthetic data. We will start by introducing the N-dimensional array (ndarray), MXNet's primary tool for storing and transforming data. 
+If you have worked with NumPy's ndarray before, you will notice that the MXNet's ndarray is an extension to the former. However, the latter provides a few key advantages. First, it supports asynchronous computation on CPU, GPU, and distributed cloud architectures, whereas the former only supports CPU computation. Second, MXNet's ndarray supports for automatic differentiation. These properties make MXNet's ndarray indispensable for deep learning.
+
 
 ## Getting Started
 
-Throughout this chapter, we are aiming to get you up and running with the basic functionality. Do not worry if you do not understand all of the basic math, like element-wise operations or normal distributions. In the next two chapters we will take another pass at the same material, teaching the material in the context of practical examples. On the other hand, if you want to go deeper into the mathematical content, see :numref:`chapter_math`.
+Throughout this chapter, we are aiming to get you up and running with the basic functionality. Do not worry if you do not understand all of the basic math, like element-wise operations or normal distributions. In the next two chapters we will take another pass at the same material with practical examples. On the other hand, if you want to go deeper into the mathematical content, see :numref:`chapter_math`.
 
-We begin by importing MXNet and the `ndarray` module from MXNet.
-Here, `nd` is short for `ndarray`.
+We begin by importing `np` module and `npx` module from MXNet. Here, `np` module includes all the functions that NumPy has, while `npx` module contains all the other extended functions which empower deep learning computation.
 
 ```{.python .input  n=1}
 from mxnet import np, npx
 npx.set_np()
 ```
 
-NDArrays represent (possibly multi-dimensional) arrays of numerical values. NDArrays with one axis correspond (in math-speak) to *vectors*. NDArrays with two axes correspond to *matrices*. For arrays with more than two axes, mathematicians do not have special names---they simply call them *tensors*.
+An ndarray represents (possibly multi-dimensional) arrays of numerical values. An ndarray with one axis corresponds (in math-speak) to *vectors*. An ndarray with two axes corresponds to *matrices*. For arrays with more than two axes, mathematicians do not have special names---they simply call them *tensors*.
 
```

Comment:
When using ndarray, we almost always invoke the `set_up` function: this is for compatibility of ndarray processing by other components of MXNet.